src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitTest.java

```diff
@@ -72,6 +76,26 @@ public void nonProtectedCheckMethodsTest() {
                 .doNotHaveModifier(JavaModifier.ABSTRACT)
                 .should().notBeProtected();
 
-        checkMethodsShouldNotBeProtectedRule.check(importedClasses);
+        checkMethodsShouldNotBeProtectedRule.check(CHECKS_PACKAGE);
+    }
+
+    /**
+     * Tests that all checks either extend {@link AbstractCheck} or {@link AbstractFileSetCheck}.
+     *
+     * @noinspection JUnitTestMethodWithNoAssertions
+     * @noinspectionreason JUnitTestMethodWithNoAssertions - asserts in callstack,
+     *     but not in this method
+     */
+    @Test
+    public void testChecksShouldExtendAbstractCheckOrAbstractFileSetCheck() {
+        final ArchRule checksShouldExtendAbstractCheckOrAbstractFileSetCheck = classes()
+            .that()
+            .haveSimpleNameEndingWith("Check")
+            .should()
+            .beAssignableTo(AbstractCheck.class)
+            .orShould()
+            .beAssignableTo(AbstractFileSetCheck.class);
```

Comment:
I do not see list of suppression. As we have several abstract classes and unfortunately they are in use.
We should check direct extension , without any classes in the middle.


Without this strict rule, this validation is useless, as it is already true and restricted by compilation

Reply:
I corrected it, done.